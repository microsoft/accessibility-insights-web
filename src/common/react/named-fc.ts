// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactFCWithDisplayName<P = {}> = React.FC<React.PropsWithChildren<P>> & {
    displayName: string;
};

export function NamedFC<P = {}>(
    displayName: string,
    component: React.FC<React.PropsWithChildren<P>>,
): ReactFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactFCWithDisplayName<P>;
}
