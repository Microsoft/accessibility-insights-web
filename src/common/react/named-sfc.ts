// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactSFCWithDisplayName<P = {}> = React.SFC<P> & { displayName: string };

export function NamedSFC<P = {}>(displayName: string, component: React.SFC<P>): ReactSFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactSFCWithDisplayName<P>;
}
