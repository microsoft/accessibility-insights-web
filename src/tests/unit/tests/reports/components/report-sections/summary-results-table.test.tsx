// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';
import {
    SummaryResultsTable,
    TableColumn,
} from 'reports/components/report-sections/summary-results-table';

describe(SummaryResultsTable.displayName, () => {
    it('renders', () => {
        const columns: TableColumn[] = [
            {
                header: 'heading1',
                contentType: 'text',
            },
            {
                header: 'heading2',
                contentType: 'url',
            },
            {
                header: 'heading3',
                contentType: 'text',
            },
        ];
        const rows = [
            ['cell1', 'cell2', <div key="cell3">cell3</div>],
            ['cell4', 'cell5', <div key="cell6">cell6</div>],
        ];
        const wrapped = shallow(
            <SummaryResultsTable columns={columns} rows={rows} id="table-id" />,
        );
        expect(wrapped.getElement()).toMatchSnapshot();
    });
});
