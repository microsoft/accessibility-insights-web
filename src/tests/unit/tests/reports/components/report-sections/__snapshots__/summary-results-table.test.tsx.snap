// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryResultsTable renders 1`] = `
<DocumentFragment>
  <table
    class="summaryResultsTable"
    id="table-id"
  >
    <thead>
      <tr>
        <th
          id="table-id-header0"
        >
          heading1
        </th>
        <th
          id="table-id-header1"
        >
          heading2
        </th>
        <th
          id="table-id-header2"
        >
          heading3
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="textCell"
          headers="table-id-header0"
        >
          cell1
        </td>
        <td
          class="urlCell"
          headers="table-id-header1"
        >
          cell2
        </td>
        <td
          class="textCell"
          headers="table-id-header2"
        >
          <div>
            cell3
          </div>
        </td>
      </tr>
      <tr>
        <td
          class="textCell"
          headers="table-id-header0"
        >
          cell4
        </td>
        <td
          class="urlCell"
          headers="table-id-header1"
        >
          cell5
        </td>
        <td
          class="textCell"
          headers="table-id-header2"
        >
          <div>
            cell6
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
